pkg/pointerdb/pointerdb.go

```diff
@@ -100,6 +100,35 @@ func (s *Server) validateSegment(req *pb.PutRequest) error {
 	return nil
 }
 
+func (s *Server) filterValidPieces(pointer *pb.Pointer) error {
+	if pointer.Type == pb.Pointer_REMOTE {
+		var remotePieces []*pb.RemotePiece
+		remote := pointer.Remote
+		for _, piece := range remote.RemotePieces {
+			err := auth.VerifyMsg(piece.Hash, piece.NodeId)
+			if err == nil {
+				// set to nil after verification to avoid storing in DB
```

Comment:
Not liking that we are conflating the message and the way we store things... but I guess it's fine for now.